### PR TITLE
scg-news-2

### PR DESCRIPTION
**Pull request by SpencerCGriffiths**

### CODE REVIEW

File: app/__tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+beforeEach(() => {
+    return seed(data);
+})
+afterAll(() => {
+    return db.end();
+})
+
+describe("GET /topics", () => {
+    describe("-- functionality tests", () => {
+        test("200: responds with a status of 200 when there is an appropriate request", () => {
```
Comment (MRombi) [inline app/__tests__/app.test.js]: We should not have a test just to check the status as we check that in every other test 🙂

File: app/__tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            return request(app)
+            .get("/api/topics")
+            .expect(200)
+        })
+        test("200: responds with an array of topic objects", () => {
+            return request(app)
+            .get("/api/topics")
+            .expect(200)
+            .then(({body}) => {
+                expect(Array.isArray(body.topics)).toBe(true)
```
Comment (MRombi) [inline app/__tests__/app.test.js]: We should not test for this if we later use anything that would imply that we are testing an array 🙂

File: app/__tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            })
+        })
+        test("200: responds with an array of topic objects with properties slug and description", () => {
+            return request(app)
+            .get("/api/topics")
+            .expect(200)
+            .then(({body}) => {
+                expect(body.topics[0]).toHaveProperty(['description']);
```
Comment (MRombi) [inline app/__tests__/app.test.js]: We should check every object inside the array, to do that we could use a loop 🙂

File: app/app.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const { getAllTopics, fourOhFour } = require("./controllers/controllers");
+const { handlePsqlErrors, handleCustomErrors, handleServerErrors } = require("./errors");
+
+
+const app = express()
+app.use(express.json())
```
Comment (MRombi) [inline app/app.js]: We do not need this at this stage and should remove it 🙂

**Review by MRombi — CHANGES_REQUESTED** [review]
Good job on your initial attempt, please do action on the feedback I left and resubmit 🙂

File: app/__tests__/app.test.js [hunk 5]
```diff
@@ -13,29 +13,18 @@ afterAll(() => {
...
-        })
-        test("200: responds with an array of topic objects with properties slug and description", () => {
+        test("200: responds with a 200 status codea and an array of topic objects with appropriate key value pairs", () => {
             return request(app)
             .get("/api/topics")
             .expect(200)
             .then(({body}) => {
-                expect(body.topics[0]).toHaveProperty(['description']); 
-                expect(body.topics[0]).toHaveProperty(['slug']); 
+                body.topics.forEach( (topic) =>{
```
Comment (aloush-dev) [inline app/__tests__/app.test.js:22]: Nice, but if you are going to forEach through the return array then we need to check the length of the array. If an empty array was returned here we would get a false passing test.

**Review by aloush-dev — APPROVED** [review]
Nice job please add a length check before merging.

**Pull request merged**